sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -411,4 +324,277 @@ public void run() {
+        this.reportOnlySampled(getProperty(KEY_SAMPLED, configMap, this.sampled));
+        return this;
+      }
+
+      /**
+       * Sets the configuration values from environment variables. This method looks for the
+       * following keys:
+       *
+       * <ul>
+       *   <li>{@code OTEL_BSP_SCHEDULE_DELAY}: to set the delay interval between two consecutive
+       *       exports.
+       *   <li>{@code OTEL_BSP_MAX_QUEUE}: to set the maximum queue size.
+       *   <li>{@code OTEL_BSP_MAX_EXPORT_BATCH}: to set the maximum batch size.
+       *   <li>{@code OTEL_BSP_EXPORT_TIMEOUT}: to set the maximum allowed time to export data.
+       *   <li>{@code OTEL_BSP_REPORT_SAMPLED}: to set whether only sampled spans should be
+       *       reported.
+       * </ul>
+       *
+       * @return this.
+       */
+      public Builder readEnv() {
+        return fromConfigMap(System.getenv(), NamingConvention.ENV_VAR);
+      }
+
+      /**
+       * Sets the configuration values from system properties. This method looks for the following
+       * keys:
+       *
+       * <ul>
+       *   <li>{@code otel.bsp.schedule.delay}: to set the delay interval between two consecutive
+       *       exports.
+       *   <li>{@code otel.bsp.max.queue}: to set the maximum queue size.
+       *   <li>{@code otel.bsp.max.export.batch}: to set the maximum batch size.
+       *   <li>{@code otel.bsp.export.timeout}: to set the maximum allowed time to export data.
+       *   <li>{@code otel.bsp.report.sampled}: to set whether only sampled spans should be
+       *       reported.
+       * </ul>
+       *
+       * @return this.
+       */
+      public Builder readSystemProperties() {
+        return fromConfigMap(Maps.fromProperties(System.getProperties()), NamingConvention.DOT);
+      }
+
+      /**
+       * Set whether only sampled spans should be reported.
+       *
+       * @param sampled report only sampled spans.
+       * @return this.
+       */
+      public Builder reportOnlySampled(boolean sampled) {
+        this.sampled = sampled;
+        return this;
+      }
+
+      /**
+       * Sets the delay interval between two consecutive exports. The actual interval may be shorter
+       * if the batch size is getting larger than {@code maxQueuedSpans / 2}.
+       *
+       * <p>Default value is {@code 5000}ms.
+       *
+       * @param scheduleDelayMillis the delay interval between two consecutive exports.
+       * @return this.
+       */
+      public Builder setScheduleDelayMillis(long scheduleDelayMillis) {
+        Utils.checkArgument(
+            scheduleDelayMillis >= 0, "scheduleDelayMillis must greater than or equal 0.");
+        this.scheduleDelayMillis = scheduleDelayMillis;
+        return this;
+      }
+
+      /**
+       * Sets the maximum time an exporter will be allowed to run before being cancelled.
+       *
+       * <p>Default value is {@code 30000}ms
```

Comment:
can we add a `@see` that links to the constant, just in case things get out of synch? Also for the other properties.